.github/workflows/ci.yml

```diff
@@ -0,0 +1,100 @@
+name: CI
+on: [pull_request]
+env:
+  RAILS_ENV: test
+  DATABASE_URL: postgresql://postgres:@localhost/test
+  DATABASE_CLEANER_ALLOW_REMOTE_DATABASE_URL: true
+jobs:
+  rubocop:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.6
+      - name: Setup
+        run: |
+          gem install bundler
+          bundle install --jobs=3 --retry=3
+      - name: Run rubocop
+        run: bundle exec rubocop
+  test_ruby:
+    runs-on: ubuntu-latest
+    needs: rubocop
+    services:
+      postgres:
+        image: postgres:12.1
+        ports: ['5432:5432']
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+    strategy:
+      fail-fast: false
+      matrix:
+        foreman-core-branch: [develop]
+        ruby-version: [2.5, 2.6]
+        node-version: [12]
+    steps:
+      - run: sudo apt-get install build-essential libcurl4-openssl-dev zlib1g-dev libpq-dev
+      - uses: actions/checkout@v2
+        with:
+          repository: theforeman/foreman
+          ref: ${{ matrix.foreman-core-branch }}
+      - uses: actions/checkout@v2
+        with:
+          path: foreman-tasks
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+      - name: Setup Node
+        uses: actions/setup-node@v1
+        with:
+          node-version:  ${{ matrix.node-version }}
+      - uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-fgems-${{ matrix.ruby-version }}-${{ hashFiles('Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-fgems-${{ matrix.ruby-version }}-
+      - name: Setup Bundler
+        run: |
+          echo "gem 'foreman-tasks', path: './foreman-tasks'" > bundler.d/foreman-tasks.local.rb
+          gem install bundler
+          bundle config set without journald development console libvirt
+          bundle config set path vendor/bundle
+      - name: Prepare test env
+        run: |
+          bundle install --jobs=3 --retry=3
+          bundle exec rake db:create
+          bundle exec rake db:migrate
+      - name: Run plugin tests
+        run: |
+          bundle exec rake test:foreman-tasks
+          bundle exec rake test TEST="test/unit/foreman/access_permissions_test.rb"
+  test_js:
+    runs-on: ubuntu-latest
+    needs: rubocop
```

Comment:
can you add paths here, so like in [core](https://github.com/theforeman/foreman/blob/develop/.github/workflows/plugins_react_tests.yml#L6) it will run only when the `webpack` folder etc.. have changed

Reply:
not as easy to do it on the jobs level. I would need to create another workflow for UI testing, or use something like https://github.com/dorny/paths-filter

Reply:
Right, want to move it to a `js_ci.yml` workflow?

Reply:
not really xD now we have to manage two matrixes of versions to test, but done so. Take a look and let me know what you think :)